### PR TITLE
fix(libsinsp): fix possible segv in parse_connect_exit + cleanups

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

ERROR: AddressSanitizer: SEGV on unknown address 0x000000001578 ... The signal is caused by a WRITE memory access. #0 0x56074dd65dc8 in sinsp_parser::parse_connect_exit(sinsp_evt*) /build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/userspace/libsinsp/parsers.cpp:3361

Plus refactor few parsers to feature early returns if evt->m_tinfo is a nullptr

Remove some inconsistent debug ASSERT statements

@LucaGuerra quick win when I ran the latest debug build on a test server just to verify if everything is good to go for further testing.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libsinsp/parsers.cpp [hunk 1]
```diff
@@ -978,7 +978,6 @@ void sinsp_parser::parse_clone_exit_caller(sinsp_evt *evt, int64_t child_tid)
 	if(caller_tinfo == nullptr)
 	{
 		/* Invalidate the thread info associated with this event */
-		ASSERT(false);
```
Comment (incertum) [inline userspace/libsinsp/parsers.cpp:981]: For the most part we don't have these and other ASSERT statements. Therefore I considered them legacy code and removed them in order to enforce a more consistent style.

File: userspace/libsinsp/parsers.cpp [hunk 2]
```diff
@@ -3343,13 +3336,18 @@ void sinsp_parser::parse_connect_exit(sinsp_evt *evt)
 	int64_t fd;
 	bool force_overwrite_stale_data = false;
 
-	if(evt->m_fdinfo == NULL)
+	if(evt->m_tinfo == nullptr)
```
Comment (incertum) [inline userspace/libsinsp/parsers.cpp:3339]: This was the actual bug.

File: userspace/libsinsp/parsers.cpp [hunk 3]
```diff
@@ -4888,33 +4882,36 @@ void sinsp_parser::parse_fcntl_exit(sinsp_evt *evt)
 
 void sinsp_parser::parse_context_switch(sinsp_evt* evt)
 {
-	if(evt->m_tinfo != nullptr)
+	if(evt->m_tinfo == nullptr)
```
Comment (incertum) [inline userspace/libsinsp/parsers.cpp:4885]: For few instances where I felt it was straight forward to do so I changed the logic to "early return" style.

**Comment by incertum** [comment]
Anyone having more context on the failed test `sinsp_with_test_input.spawn_process_container`? It seems to be a newer test.

File: userspace/libsinsp/parsers.cpp (hunk 1 above)
Comment (LucaGuerra) [reply userspace/libsinsp/parsers.cpp:981]: Thank you. We need to get rid of legacy asserts!

**Review by LucaGuerra — APPROVED** [review]
LGTM

**Pull request merged**